src/widgets/modals/StocktakeBatchModal.js

```diff
@@ -89,14 +96,26 @@ export const StocktakeBatchModalComponent = ({
     ? pageColumns.filter(({ key }) => key !== COLUMN_KEYS.SNAPSHOT_TOTAL_QUANTITY)
     : pageColumns;
 
-  const { stocktake = {} } = stocktakeItem;
+  const { stocktake = {}, isVaccine = false } = stocktakeItem;
   const { isFinalised = false } = stocktake;
   const { difference = 0 } = modalValue || {};
 
-  const reasonsSelection =
-    difference > 0
-      ? UIDatabase.objects('PositiveAdjustmentReason')
-      : UIDatabase.objects('NegativeAdjustmentReason');
+  let reasonsSelection = null;
+  if (usingReasons && !isVaccine) {
+    reasonsSelection =
+      difference > 0
+        ? UIDatabase.objects('PositiveAdjustmentReason')
+        : UIDatabase.objects('NegativeAdjustmentReason');
+  } else if (usingOpenVialWastageReasons && isVaccine) {
```

Comment:
It is possible that `usingReasons` is false and it is a vaccine and there aren't any positive adjustment reasons, only the wastage reasons.

Though it really doesn't matter as the `REASON` column only shows if there are positive and negative reasons available. Really just wondering if the query could be simplified at all.

Reply:
Yeah, it's not great and I don't know what to do about it.. :(